### PR TITLE
Assignment 5

### PR DESCRIPTION
**Pull request by wpmcgrath95**
I think most of the logic is there, however, I get a very weird error when running mariaDB. It just stops here: 
"db-container  | Version: '10.5.8-MariaDB-1:10.5.8+maria~focal'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution." 

I tried fixing this for a while and I can't seem to get rid of the bug. Others have had the same problem but can't fix it: https://stackoverflow.com/questions/62922399/docker-compose-mariadb-docker-entrypoint-initdb-d-sql-is-not-executed. 

Any help would be greatly appreciated! Thank you.

### CODE REVIEW

**Comment by gramadin** [comment]
When i had this error i ran the container in regular mode (not detached) then used 
`docker exec -it db-container bin/bash\`
this got me into the container cli
then i would look for make sure the files you are trying to work with are actually there.

I found that I though i was copying a sql file, but instead i was just making a folder with the name of the file i wanted to copy

after that i resolved the file copy and was able to get the sql to copy and load manually.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+version: "3.3"
+services:
```
Comment (gramadin) [inline docker-compose.yml:2]: I saw you are using a python method to get and display the bb report, but you don't set up the service here?

File: SQL_files/baseball_batting_average.sql [hunk 2]
```diff
@@ -58,4 +58,32 @@ JOIN baseball.BATTERS_ROLLING AS b
 ON a.local_date WHERE DATEDIFF(a.local_date,b.local_date) BETWEEN 0 AND 100 AND a.batter=b.batter
 ORDER BY a.local_date ASC 
 LIMIT 0,20; # use join on itself instead of 2nd select
-*/
\ No newline at end of file
+*/
```
Comment (gramadin) [inline SQL_files/baseball_batting_average.sql:61]: Do you need both sql files here?

File: python_files/create_html_table.py [hunk 3]
```diff
@@ -8,7 +8,6 @@
 class CreateHTMLTable(object):
```
Comment (gramadin) [inline python_files/create_html_table.py:8]: not sure where you are using this file

File: python_files/server.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+from flask import Flask
```
Comment (gramadin) [inline python_files/server.py:1]: Maybe this is not the right file version?

File: python_files/create_html_table.py (hunk 3 above)
Comment (wpmcgrath95) [reply python_files/create_html_table.py:8]: oh this was for the midterm lol

File: SQL_files/baseball_batting_average.sql (hunk 2 above)
Comment (wpmcgrath95) [reply SQL_files/baseball_batting_average.sql:61]: I just had all my stuff from pervious hws so when I merge there isn't any conflicts

File: docker-compose.yml (hunk 1 above)
Comment (wpmcgrath95) [reply docker-compose.yml:2]: I thought setting up a service was made once you name it as I did below " db_service." Is that not right?

**Comment by wpmcgrath95** [comment]
> When i had this error i ran the container in regular mode (not detached) then used
> `docker exec -it db-container bin/bash\`
> this got me into the container cli
> then i would look for make sure the files you are trying to work with are actually there.
> 
> I found that I though i was copying a sql file, but instead i was just making a folder with the name of the file i wanted to copy
> 
> after that i resolved the file copy and was able to get the sql to copy and load manually.

Oh ok I'll try that. Did you have to manually enter these commands or can this work with a bash script?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# base image
+FROM mariadb:latest
+
+# set the working directory in the container
+ENV APP_HOME /app
+WORKDIR $APP_HOME
+ENV PYTHONPATH /
+
+# get necessary python libraries
+COPY requirements.txt .
```
Comment (bwilder95) [inline Dockerfile:10]: Are the python libraries actually required here? I had a similar question on Ed's too. I don't think I had to use any for mine.

**Comment by bwilder95** [comment]
Hi !!

Sorry can someone share the link for the class! I switched to a linux OS
and forgot to bring in my zoom thing .

Thanks ,
Brent

**Pull request merged**